### ISSUE TITLE
Vvariable 'b' is uninitialized when used here [-Wuninitialized]

### ISSUE THREAD

**Issue by yewentao256** [darglein/TinyTorch#2]
Hello, in TinyTorch-main/src/optimizer.h:45

```c++
                // sgd with nesterov momentum
                float b;
                if (step > 0)
                {
                    b = momentum * b + (1 - 0.1) * g;

                    g = g + momentum * b;
                }

                w = w - lr * g;
```

the variable `b` is used without initialization, will this cause a problem?

When I try to print b like this

```c++
                float b;
                if (step == 0){
                    std::cout << "uninitialized b:" << b << std::endl;
                }
```

The compiler complains about: warning: variable 'b' is uninitialized when used here [-Wuninitialized]
            std::cout << "uninitialized b:" << b << std::endl;
and the value is not expected: 

```
uninitialized b:1.4013e-45
uninitialized b:1.4013e-45
uninitialized b:1.4013e-45
uninitialized b:1.4013e-45
...
```

So is this a bug? Thanks and looking forward to your reply

And another question: why there is a `(1.0 - 0.1) * g`, does this const number mean learning rate?

**Comment by darglein** [comment]
Hey, 

thanks for the issue. I have fixed this problem in the latest commit. The term `(1.0 - 0.1) * g`  is the dampening from the pytorch implementation with a dampening value of 0.1. I have also renamed it to match the pytorch implementation from here:

https://pytorch.org/docs/stable/generated/torch.optim.SGD.html

**Comment by yewentao256** [comment]
Thanks a lot~
